client-java/src/main/java/uk/gov/gchq/palisade/client/receiver/AbstractReceiver.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2020 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.palisade.client.receiver;
+
+/**
+ * Super class of receivers
+ *
+ * @since 0.5.0
+ */
+public abstract class AbstractReceiver implements Receiver {
+
+    /**
+     * Property key for the number of bytes downloaded
+     */
+    public static final String BYTES_KEY = "bytes";
+
+    /**
+     * Property key for the path where downloads are saved
+     */
+    public static final String PATH_KEY = "path";
+
+    /**
+     * Property key for the filename
+     */
+    public static final String FILENAME_KEY = "filename";
+
+    /**
+     * Create the instance
+     */
+    @SuppressWarnings("java:S1118")
+    public AbstractReceiver() {
```

Comment:
This one caused me pause. I did have the static strings in the `Receiver` interface, but sonar rightly complained. Even Josh Bloch says don't do that. So I created the `AbstractReceiver`. But I'm still not sure you want other developers extending it. They would probably ignore it and implement Receiver interface anyway. 

I'll think about how to resolve this one. It could be that we provide the Receiver class on the `client.submit()` call. We could then return a targeted return value. But this would then limit the return values. Hmmmm, decisions.

Reply:
fixed